lib/clients/AzureClient.py

```diff
@@ -94,26 +102,50 @@ def get_volume(self, volume_name):
                     volume_name, error))
             return None
 
+    def get_hot_number_of_data_volumes(self):
+        '''
+        This particual funtion is specific for Azure.
+        This determines the scsi host number for the persistent disk.
+        The host number along with lun would be required to find out device deterministic way.
+        '''
+        host_number = None
+        try:
+            device_persistent_volume = self.shell(
+                'cat {} | grep {}'
+                .format(self.FILE_MOUNTS, self.DIRECTORY_PERSISTENT)).split(' ')[0][5:-1]
+            device_paths = glob.glob(
+                '/sys/bus/scsi/devices/*:*:*:*/block/{}'.format(device_persistent_volume))
+            if len(device_paths) > 1:
+                raise Exception('Invalid device paths for device {}'.format(
+                    device_persistent_volume))
+            # Success: Go only one device path
+            host_number = device_paths[0][22:-len(
+                '/block/{}'.format(device_persistent_volume))].split(':')[0]
+        except Exception as error:
+            self.logger.error(
+                '[ERROR] [SCSI HOST NUMBER] [DATA VOLUME] Error while determining SCSI host number'
+                'of persistent volume directory {}.{}'.format(self.DIRECTORY_PERSISTENT, error))
+        return host_number
+
     def get_attached_volumes_for_instance(self, instance_id):
         instance = self.compute_client.virtual_machines.get(
             self.resource_group, instance_id
         )
         try:
-            volumeList = []
+            volume_list = []
             for disk in instance.storage_profile.data_disks:
                 device = None
-                if disk.lun == 0:
-                    device = self.shell(
-                        'cat {} | grep {} | grep /dev/ '
-                        .format(self.FILE_MOUNTS, self.DIRECTORY_DATA)).split(' ')[0][:8]
-                elif disk.lun == 1:
-                    device = self.shell(
-                        'cat {} | grep {}'
-                        .format(self.FILE_MOUNTS, self.DIRECTORY_PERSISTENT)).split(' ')[0][:8]
-
-                volumeList.append(
+                device_path = glob.glob(
+                    '/sys/bus/scsi/devices/{}:*:*:{}/block'.format(self.scsi_host_number, disk.lun))
```

Comment:
can put this path into constants , used above as well .

Reply:
and below as well.

Reply:
Done. Updating.